apps/src/turtle/blocks.js

```diff
@@ -1096,73 +1096,83 @@ exports.install = function (blockly, blockInstallOptions) {
         '(\'block_id_' + this.id + '\');\n';
   };
 
-  // We alias 'turtle_stamp' to be the same as the 'sticker' block for
-  // backwards compatibility.
-  blockly.Blocks.sticker = blockly.Blocks.turtle_stamp = {
-    helpUrl: '',
-    init: function () {
-      this.setHSV(184, 1.00, 0.74);
-      var dropdown;
-      var input = this.appendDummyInput();
-      input.appendTitle(msg.drawSticker());
-
-      // Generates a list of pairs of the form [[url, name]]
-      var values = [];
-      for (var name in skin.stickers) {
-        var url = skin.stickers[name];
-        values.push([url, name]);
-      }
+  function createDrawStickerBlock(blockName)  {
+    return {
+      helpUrl: '',
+      init: function () {
+        this.setHSV(184, 1.00, 0.74);
+        var dropdown;
+        var input = this.appendDummyInput();
+        input.appendTitle(msg.drawSticker());
+        this.setInputsInline(true);
+        this.setPreviousStatement(true);
+        this.setNextStatement(true);
+
+        // Generates a list of pairs of the form [[url, name]]
+        var values = [];
+        for (var name in skin.stickers) {
+          var url = skin.stickers[name];
+          values.push([url, name]);
+        }
+        dropdown = new blockly.FieldImageDropdown(values, 40, 40);
 
-      dropdown = new blockly.FieldImageDropdown(values, 40, 40);
+        input.appendTitle(dropdown, 'VALUE');
 
-      input.appendTitle(dropdown, 'VALUE');
+        appendToDrawStickerBlock(blockName, this);
+      }
+    };
+  }
 
-      this.setInputsInline(true);
-      this.setPreviousStatement(true);
-      this.setNextStatement(true);
-      this.setTooltip(msg.drawSticker());
+  // Add size input to the draw sticker block (text input & socket)
+  function appendToDrawStickerBlock(blockName, block) {
+    if (blockName === 'turtle_sticker_with_size') {
+      block.appendDummyInput()
+           .appendTitle(msg.withSize());
+      block.appendValueInput('SIZE')
+           .setCheck(blockly.BlockValueType.NUMBER);
+      block.appendDummyInput()
+           .appendTitle(msg.pixels());
+      block.setTooltip(msg.drawStickerWithSize());
+    } else if (blockName === 'turtle_sticker_with_size_non_param') {
+      block.appendDummyInput()
+           .appendTitle(msg.withSize());
+      block.appendDummyInput()
+           .appendTitle(new blockly.FieldTextInput('0',
+           blockly.FieldTextInput.numberValidator), 'SIZE')
```

Comment:
Since this line is a continuation of the function call on the previous line rather than one of the chained method calls, it should be indented an additional 4 spces.